src/jaxsim/mujoco/loaders.py

```diff
@@ -598,21 +611,106 @@ def convert(
-            raise ValueError("xyaxes must have six values separated by space")
+        Note:
+            The choice of the parameters is easier if we imagine to consider a target
+            frame `T` whose origin is located over the lookat point and having the same
+            orientation of the world frame `W`. We also introduce a camera frame `C`
+            whose origin is located over the lower-left corner of the image, and having
+            the x-axis pointing right and the y-axis pointing up in image coordinates.
+            The camera renders what it sees in the -z direction of frame `C`.
 
-        return cls(**kwargs)
+        Args:
+            camera_name: The name of the camera.
+            lookat: The target point to look at (origin of `T`).
+            distance:
+                The distance from the target point (displacement between the origins
+                of `T` and `C`).
+            azimut:
+                The rotation around z of the camera. With an angle of 0, the camera
+                would loot at the target point towards the positive x-axis of `T`.
+            elevation:
+                The rotation around the x-axis of the camera frame `C`. Note that if
+                you want to lift the view angle, the elevation is negative.
+            degrees: Whether the angles are in degrees or radians.
+            **kwargs: Additional camera parameters.
+
+        Returns:
+            The custom camera.
+        """
+
+        # Start from a frame whose origin is located over the lookat point.
+        # We initialize a -90 degrees rotation around the z-axis because due to
+        # the default camera coordinate system (x pointing right, y pointing up).
+        W_H_C = np.eye(4)
+        W_H_C[0:3, 3] = np.array(lookat)
+        W_H_C[0:3, 0:3] = Rotation.from_euler(
+            seq="Z", angles=-90, degrees=degrees
+        ).as_matrix()
+
+        # Process the azimut.
+        R_az = Rotation.from_euler(seq="Z", angles=azimut, degrees=degrees).as_matrix()
+        W_H_C[0:3, 0:3] = W_H_C[0:3, 0:3] @ R_az
+
+        # Process elevation.
+        R_el = Rotation.from_euler(
+            seq="X", angles=-elevation, degrees=degrees
+        ).as_matrix()
+        W_H_C[0:3, 0:3] = W_H_C[0:3, 0:3] @ R_el
+
+        # Process distance.
+        tf_distance = np.eye(4)
+        tf_distance[2, 3] = distance
+        W_H_C = W_H_C @ tf_distance
+
+        # Extract the position and the quaternion.
+        p = W_H_C[0:3, 3]
+        Q = Rotation.from_matrix(W_H_C[0:3, 0:3]).as_quat()[np.array([3, 0, 1, 2])]
+
+        return MujocoCamera.build(
+            name=camera_name,
+            mode="fixed",
+            fovy="60",
+            pos=" ".join(p.astype(str).tolist()),
+            quat=" ".join(Q.astype(str).tolist()),
+            **kwargs,
+        )
+
+    def asdict(self) -> dict[str, str]:
+
+        return {k: v for k, v in dataclasses.asdict(self).items() if v is not None}
```

Comment:
Wouldn't it be a bit confusing to silently exclude the `None` values? Probably this logic can fit better in the `__repr__` instead

Reply:
I wrote this method before adding to [`RodModelToMjcf.convert`](https://github.com/ami-iit/jaxsim/blob/e9a89e32f45aff03bef48e27ac8be3e28b592b71/src/jaxsim/mujoco/loaders.py#L170-L171) the support of receiving `MujocoCamera|Sequence[MujocoCamera]`. Before this update, the method was taking a dictionary that, if generated using `dataclasses.asdict` from a `MujocoCamera` object, included the `None` elements. The conversion to XML of such dictionary was failing, the None attributes had to be ignored.

What do you suggest? I can remove this method now, it's no longer needed.

Reply:
Yes, I guess that users can eventually use `dataclasses.asdict` for this

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/206/commits/6b86a1527072578eca4a26863b45526bf7aef228.

Reply:
> > I can remove this method now, it's no longer needed.
> 
> Yes, I guess that users can eventually use `dataclasses.asdict` for this

Note that if there are `None` attributes, the dictionary produced with `dataclasses.asdict` won't be valid when fed into `lxml` since it would include in the XML attributes with wrong content not supported by the schema.

Reply:
Woops forget everything I wrote. I forgot I was using this method [here](https://github.com/ami-iit/jaxsim/blob/f0e6f0c1b075e664cfcbc1236729ba6035594d09/src/jaxsim/mujoco/loaders.py#L499). Brought it back.